go/vt/vttablet/tabletmanager/vreplication/engine.go

```diff
@@ -257,6 +258,7 @@ func (vre *Engine) retry(ctx context.Context, err error) {
 
 func (vre *Engine) initControllers(rows []map[string]string) {
 	for _, row := range rows {
+		vre.repairRow(row)
```

Comment:
We need to change the point in the tablet lifecycle where we do this repair. 

I tested your branch manually by running the local example upto and including the `./202_move_tables.sh` step after modifying that script to use 
`vtctlclient MoveTables -tablet_types=master -source commerce -tables 'customer,corder' Create customer.commerce2customer`. I then ran
`PlannedReparentShard -keyspace_shard=customer/0 -new_primary=zone1-0000000201 -wait_replicas_timeout=1m`. However the PRS fails because the new primary does not start up because of the repair query. I see this in the logs for 201.

```
I0111 14:18:31.323194   28084 engine.go:284] Running query update _vt.vreplication set tablet_types='PRIMARY' where id=1
I0111 14:18:33.120948   28084 state_manager.go:628] Going unhealthy due to replication error: no replication status (errno 100) (sqlstate HY000)
```

I see from `updateLocked()` in `tm_state.go` that the serving type is only set after the vreplication engine is opened. So we need to run this repair later in the cycle. Maybe we can just do this in sql and add it to the `WithDDLInitialQueries` list at the end because those queries will only run once the tablet is serving.

Reply:
I changed the implementation to add the repair queries at the beginning of `withDDLInitialQueries`. However, I had to do three versions because I could not use `UPPER(tablet_types)`, since `tablet_types` is `varbinary` not `varchar`. Let me know if there is a better way of doing it.

Reply:
I tested the local example with your steps and made sure that `PlannedReparentShard` now succeeds. I temporarily added a log line to `controller.go` to print out the queries as they're executed.
Before PRS:
```
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.321005   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'MASTER', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.321407   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.322924   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'Master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.323274   81291 controller.go:202] Executing init query 'SELECT db_name FROM _vt.vreplication LIMIT 0'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.323413   81291 controller.go:202] Executing init query 'SELECT rows_copied FROM _vt.vreplication LIMIT 0'
```

After PRS:
```
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.321005   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'MASTER', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.321407   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.322924   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'Master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.323274   81291 controller.go:202] Executing init query 'SELECT db_name FROM _vt.vreplication LIMIT 0'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:05:41.323413   81291 controller.go:202] Executing init query 'SELECT rows_copied FROM _vt.vreplication LIMIT 0'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:09:22.117432   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'MASTER', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:09:22.117840   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:09:22.118138   81291 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'Master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:09:22.118384   81291 controller.go:202] Executing init query 'SELECT db_name FROM _vt.vreplication LIMIT 0'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200507.81291:I0111 20:09:22.118662   81291 controller.go:202] Executing init query 'SELECT rows_copied FROM _vt.vreplication LIMIT 0'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200512.81782:I0111 20:09:17.263371   81782 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'MASTER', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200512.81782:I0111 20:09:17.263795   81782 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200512.81782:I0111 20:09:17.263963   81782 controller.go:202] Executing init query 'UPDATE _vt.vreplication SET tablet_types=REPLACE(tablet_types, 'Master', 'PRIMARY')'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200512.81782:I0111 20:09:17.264101   81782 controller.go:202] Executing init query 'SELECT db_name FROM _vt.vreplication LIMIT 0'
vttablet.arvinds-mbp.arvind.log.INFO.20220111-200512.81782:I0111 20:09:17.264299   81782 controller.go:202] Executing init query 'SELECT rows_copied FROM _vt.vreplication LIMIT 0'
```

Reply:
> I changed the implementation to add the repair queries at the beginning of `withDDLInitialQueries`. However, I had to do three versions because I could not use `UPPER(tablet_types)`, since `tablet_types` is `varbinary` not `varchar`. Let me know if there is a better way of doing it.

What about something like `update vreplication set tablet_types = replace(lower(convert(tablet_types using utf8mb4)), 'master', 'primary') where instr(convert(tablet_types using utf8mb4), 'master');`?